File: client/src/SplitText.js

export default function SplitText(text, setAudioUrl){
    let result = [];
    const words = text.split(" ")
    let currWord = ''
    for (let i = 0; i < words.length; i += 1) {
        if (currWord.length + words[i].length < 3500){
            currWord += (words[i] + " ")
        }else{
            result.push(currWord)
            currWord = words[i]
        }
        
    }
    if (currWord){
        result.push(currWord)
    }
    console.log(result)
    return result
}

File: server/endpoint_methods/split_text.py
def split_text(text):
    result = []
    words = text.split(" ")
    curr_word = ''
    
    for word in words:
        if len(curr_word) + len(word) < 3500:
            curr_word += (word + " ")
        else:
            result.append(curr_word.strip())  # Remove the extra space at the end
            curr_word = word + " "
    
    if curr_word:
        result.append(curr_word.strip())
    
    return result
